server/src/main/java/org/apache/druid/server/http/HistoricalResource.java

```diff
@@ -50,14 +50,14 @@ public HistoricalResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Response getLoadStatus()
   {
-    return Response.ok(ImmutableMap.of("cacheInitialized", coordinator.isStarted())).build();
+    return Response.ok(ImmutableMap.of("cacheInitialized", segmentLoadDropHandler.isStarted())).build();
   }
 
   @GET
   @Path("/readiness")
   public Response getReadiness()
   {
-    if (coordinator.isStarted()) {
+    if (segmentLoadDropHandler.isStarted()) {
```

Comment:
Hmm, is this actually true if still using zk segment loading? It was wrong before for the same reasons I think for http segment loading. Maybe this resource should accept either segment loader like `SegmentListerResource`, or we should have another http resource that we bind instead depending which mode is enabled?

Reply:
primary thing checked  by this endpoint is  that  historical startup  sequence has  finished  loading/announcing segments that  it already had  on disk which  is often a time consuming activity. that is still  ensured by `segmentLoadDropHandler.isStarted()`